src/app/server/ExampleAclStorage.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/server/ExampleAclStorage.h>
+
+#include <lib/support/DefaultStorageKeyAllocator.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::Access;
+
+using EncodableEntry   = AclStorage::EncodableEntry;
```

Comment:
But isn't  using `AclStorage::EncodableEntry` something specific to this storage implementation?  That is, why is that type on AclStorage and not on this subclass?

Reply:
Those are helper classes provided here. They don't have to be used by any derived implementation. But they are used by the cluster code (which isn't involved with persistent storage, but does encode/decode ACLs). That code could be broken out into another place, but I'm in favour of simplicity.